File: src/main/java/com/sonet/storage/model/producer/Producer.java
package com.sonet.storage.model.producer;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Entity
@Table(name = "producer",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "name")
        })
@ToString
@NoArgsConstructor
@Getter
@Setter
public class Producer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Size(max = 20)
    private String name;

    @NotBlank
    @Size(max = 10)
    private String country;

    @NotBlank
    @Size(max = 200)
    private String description;

    public Producer(String name, String country, String description) {
        this.name = name;
        this.country = country;
        this.description = description;
    }
}

File: src/main/webapp/storage/src/app/dto/Producer.ts
export class Producer {
    private readonly _id: number;
    private readonly _name: string;
    private readonly _country: string;
    private readonly _description: string;


  constructor(id: number, name: string, country: string, description: string) {
    this._id = id;
    this._name = name;
    this._country = country;
    this._description = description;
  }


  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get country(): string {
    return this._country;
  }

  get description(): string {
    return this._description;
  }
}
